library/core/src/option.rs

```diff
@@ -689,6 +689,90 @@ impl<T> Option<T> {
         }
     }
 
+    const fn get_some_offset() -> isize {
+        if core::mem::size_of::<Option<T>>() == core::mem::size_of::<T>() {
+            // niche optimization means the `T` is always stored at the same position as the Option.
+            0
+        } else {
+            let some_uninit = Some(core::mem::MaybeUninit::<T>::uninit());
```

Comment:
Maybe `assert!(core::mem::size_of::<Option<T>>() == core::mem::size_of::<Option<MaybeUninit<T>>>())`, just to be extra safe in case of new optimizations or something?

Reply:
Good idea! I'll add that.